tests/functional/test_f_household.py

```diff
@@ -79,3 +83,74 @@ def test_list_household_select_all_fields(browser, household: "CountryHousehold"
     checkboxes = browser.find_elements('input[type="checkbox"][name="fields"]')
     for cb in checkboxes:
         assert cb.is_selected()
+
+
+def _test_update_with_regex(
+    browser,
+    household: "CountryHousehold",
+    link: str,
+    field: str,
+    regex: str,
+    subst: str,
+):
+    browser.login_as_user()
+    browser.select_option_by_text("select[name=tenant]", household.program.country_office.name)
+    browser.select2_select("id_program", household.program.name)
+    browser.click_link(link)
+
+    browser.click("#action-toggle")
+    browser.select_option_by_text("select[name=action]", "Update fields using RegEx")
+    browser.click("button[name='index'][value='0']")
+
+    browser.assert_element_visible("#id_field")
+    browser.assert_element_visible("#id_regex")
+    browser.assert_element_visible("#id_subst")
+
+    assert not browser.execute_script("return document.querySelector('#id_regex').checkValidity();")
+    assert not browser.execute_script("return document.querySelector('#id_subst').checkValidity();")
+
+    select_element = browser.find_element("#id_field")
+    select = Select(select_element)
+    select.select_by_value(field)
+    browser.fill("#id_regex", regex)
+    browser.fill("#id_subst", subst)
+
+    browser.click('input[name="_preview"]')
+    headers = browser.find_elements(By.XPATH, "//table//tr[1]/th")
+    header_texts = [h.text.strip().lower() for h in headers]
+    for hdr in ["pk", "old", "new"]:
+        assert hdr in header_texts
+
+    cells = browser.find_elements(By.XPATH, "//table//tr[2]/td")
+    third_cell_text = cells[3].text.strip()
+    assert subst in third_cell_text
+
+    browser.click('input[name="_apply"]')
+    browser.click('//a[div[text()="Async Jobs"]]')
+    browser.wait_for_element("table#result_list", timeout=10)
+    rows = browser.find_elements("table#result_list tbody tr")
+    assert any("Update fields using RegEx" in row.text for row in rows)
+
+
+@pytest.mark.selenium
+def test_households_update_with_regex(browser: CountryWorkspaceSeleniumTC, household: "CountryHousehold"):
```

Comment:
Let's parameterise this test using different regex. Let's check escaping and invalid regex